From: Chris Liddell <chris.liddell@artifex.com>
Date: Tue, 31 Jan 2023 10:16:45 +0000 (+0000)
Subject: Bug 706360: Catch and act on error code during PCL5 init
X-Git-Tag: rjw-test~16
X-Git-Url: https://git.ghostscript.com/?p=ghostpdl.git;a=commitdiff_plain;h=936d966618c8

Bug 706360: Catch and act on error code during PCL5 init

Missing fonts is a fatal error, and we were ignoring the error from
pcl_do_resets() due to "missing" fonts (i.e. non-romfs configs not supported).
---

diff --git a/pcl/pcl/pctop.c b/pcl/pcl/pctop.c
index 224614d5d..38732d5e2 100644
--- a/pcl/pcl/pctop.c
+++ b/pcl/pcl/pctop.c
@@ -1,4 +1,4 @@
-/* Copyright (C) 2001-2022 Artifex Software, Inc.
+/* Copyright (C) 2001-2023 Artifex Software, Inc.
    All Rights Reserved.
 
    This software is provided AS-IS with no warranty, either express or
@@ -281,8 +281,11 @@ pcl_impl_allocate_interp_instance(pl_interp_implementation_t *impl,
     /* Return success */
     impl->interp_client_data = pcli;
     /* Initial reset for the PCL interpreter */
-    pcl_do_resets(&pcli->pcs, pcl_reset_initial);
-    return 0;
+    code = pcl_do_resets(&pcli->pcs, pcl_reset_initial);
+    if (code < 0) {
+        (void)impl->proc_deallocate_interp_instance(impl);
+    }
+    return code;
 }
 
 /* if the device option string PCL is not given, the default
